pkg/provisioner/ironic/ironic.go

```diff
@@ -822,6 +822,159 @@ func (p *ironicProvisioner) getUpdateOptsForNode(ironicNode *nodes.Node, checksu
 	return updates, nil
 }
 
+// A private method to build RAID disks
+func (p *ironicProvisioner) buildTargetRAIDCfg() (logicalDisks []nodes.LogicalDisk, err error) {
+	// If not configure raid
+	if nil == p.host.Spec.RAID || nil == p.host.Spec.RAID.Volumes || 0 == len(p.host.Spec.RAID.Volumes) {
+		return
+	}
+	// Build logicalDisks
+	var logicalDisk nodes.LogicalDisk
+	var rootVolumeFlag bool = false
+	nameCheckFlags := make(map[string]bool)
+	for _, volume := range p.host.Spec.RAID.Volumes {
+		if volume.IsRootVolume {
+			if rootVolumeFlag {
+				err = errors.Errorf("most one logical disk can be set to root")
+				p.log.Error(err, "failed in build the target RAID configuration")
+				return nil, err
+			}
+			rootVolumeFlag = true
+		}
+		if nameCheckFlags[volume.Name] {
+			err = errors.Errorf("name of volume must be unique")
+			p.log.Error(err, "failed in build the target RAID configuration")
+			return nil, err
+		}
+		nameCheckFlags[volume.Name] = true
```

Comment:
Doesn't go have a set structure (like Python's `set`)?

Reply:
It seems that golang does not implement `set` in the standard library.

Reply:
Sorry, It seems that golang does not implement `set` in the standard library.